### PR TITLE
Accept invitation

### PR DESCRIPTION
**Pull request by sgiaccio**

### CODE REVIEW

File: server/modules/user/api/userApi.js [hunk 1]
```diff
@@ -60,6 +60,17 @@ module.exports.init = app => {
 
   // ==== UPDATE
 
+  app.put('/user/username', async (req, res, next) => {
+    try {
+      const { user } = req
+      const { name } = Request.getParams(req)
+
+      res.json(UserService.updateUsername(user, name))
```
Comment (minotogna) [inline server/modules/user/api/userApi.js]: don't you need to await here? would it be better to have a variable?

File: server/modules/user/api/userApi.js [hunk 2]
```diff
@@ -60,6 +60,17 @@ module.exports.init = app => {
 
   // ==== UPDATE
 
+  app.put('/user/username', async (req, res, next) => {
+    try {
```
Comment (minotogna) [inline server/modules/user/api/userApi.js]: can we use the standard api path like ```/user/:userId/username/```?

File: server/modules/user/repository/userRepository.js [hunk 3]
```diff
@@ -96,6 +107,7 @@ module.exports = {
   fetchUserByEmail,
 
   // UPDATE
+  updateUsername,
   updateUserPref,
```
Comment (minotogna) [inline server/modules/user/repository/userRepository.js]: do we still need ```selectFieldsCommaSep```?

File: webapp/login/acceptInvitationForm.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
+import React, { useRef } from 'react'
+import { connect } from 'react-redux'
+
+import { acceptInvitation, setLoginError } from './actions'
+
+const AcceptInvitationForm = props => {
+
+  const { acceptInvitation, setLoginError } = props
+
+  const usernameRef = useRef(null)
```
Comment (minotogna) [inline webapp/login/acceptInvitationForm.js]: is this ref referring to username or name?

File: webapp/login/loginView.js [hunk 5]
```diff
@@ -1,24 +1,28 @@
...
-  const usernameRef = useRef(null)
-  const passwordRef = useRef(null)
+  let form
+  switch (requiredUserAction) {
+    case LoginState.userActions.setNewPassword:
+      form = <AcceptInvitationForm />
+      break
+    default:
+      form = <LoginForm />
+  }
```
Comment (minotogna) [inline webapp/login/loginView.js]: can you just have a ternary operator in the render directly ? e.g.

```
 requiredUserAction === LoginState.userActions.setNewPassword
? <AcceptInvitationForm />
:  <LoginForm />
```

File: server/modules/user/repository/userRepository.js [hunk 6]
```diff
@@ -38,49 +37,61 @@ const fetchUsersBySurveyId = async (surveyId, offset = 0, limit = null, client =
...
+    SELECT *
+    FROM "user"
+    WHERE email = $1`,
+    [email],
+    camelize)
 
 // ==== UPDATE
 
+const updateUsername = async (user, name, client = db) =>
+  client.one(`
```
Comment (minotogna) [inline server/modules/user/repository/userRepository.js]: let's always add ```**await**```

File: server/modules/user/api/userApi.js (hunk 1 above)
Comment (sgiaccio) [reply server/modules/user/api/userApi.js]: Added await

File: server/modules/user/api/userApi.js (hunk 2 above)
Comment (sgiaccio) [reply server/modules/user/api/userApi.js]: yes

File: server/modules/user/repository/userRepository.js (hunk 6 above)
Comment (sgiaccio) [reply server/modules/user/repository/userRepository.js]: yes

**Pull request merged**